frontend/javascripts/oxalis/view/right-border-tabs/tree_hierarchy_view.tsx

```diff
@@ -157,8 +151,15 @@ function TreeHierarchyView(props: Props) {
       ).map((node) => node.key);
       expandedKeys = _.without(expandedKeys, ...subGroupKeys);
     }
-
-    setExpandedNodeKeys(expandedKeys);
+    const expandedKeySet = new Set(expandedKeys);
+    const newGroups = mapGroups(props.treeGroups, (group) => {
+      if (expandedKeySet.has(getNodeKey(GroupTypeEnum.GROUP, group.groupId))) {
+        return { ...group, isExpanded: true };
+      } else {
+        return { ...group, isExpanded: false };
+      }
+    });
```

Comment:
you can avoid creating new instances if the expanded property did not change. I think, I suggested the same for the segment groups PR. please do the same change there in case it slipped through.

```suggestion
    const newGroups = mapGroups(props.treeGroups, (group) => {
      const shouldBeExpanded = expandedKeySet.has(getNodeKey(GroupTypeEnum.GROUP, group.groupId))
      if (shouldBeExpanded !== group.isExpanded) {
        return { ...group, isExpanded: shouldBeExpanded };
      } else {
        return group;
      }
    });
```

Reply:
ah, damn. yeah, it did. I may need to fix it somewhere else, right? because that code isnt in the base branch. as I understood, master wont be merged into this branch here and I am not sure whether this branch or its base will be merged first.

Reply:
yeah, maybe simply open a new pr :)

Reply:
(for reference https://github.com/scalableminds/webknossos/pull/7947)